app/controllers/short_url_requests_controller.rb

```diff
@@ -49,6 +51,21 @@ def reject
     redirect_to short_url_requests_path
   end
 
+  def destroy
+    router.delete_route("#{@short_url_request.from_path}")
+    if router.commit_routes
+      @short_url_request.destroy
```

Comment:
I reckon we shouldn't be deleting the `short_url_request`.  Instead we should probably be setting the state to 'deleted' or something similar.  We should probably delete the `Redirect` instance though.